doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,235 @@
+<https://docs.scipy.org/doc/numpy/reference/testing.html>`_.
+
+Testing that expected exceptions are raised
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When writing a new test that a function call raises a certain exception,
+the preferred style is to use ``pytest.raises`` as a context manager, with
+the code that is supposed to raise the exception in the code block defined
+by the context manager.  The `match` keyword argument of ``pytest.raises``
+is given with enough of the expected error message attached to the exception
+to ensure that the expected exception is raised.
+
+For example, the function `scipy.stats.zmap` is supposed to raise a
+``ValueError`` if the input contains ``nan`` and ``nan_policy`` is ``"raise"``.
+A test for this is::
+
+    scores = np.array([1, 2, 3])
+    compare = np.array([-8, -3, 2, 7, 12, np.nan])
+    with pytest.raises(ValueError, match='input contains nan'):
+        stats.zmap(scores, compare, nan_policy='raise')
+
+The ``match`` argument ensures that the test doesn't pass by raising
+a ``ValueError`` that is not related to the input containing ``nan``.
+
+
+A parameter called lambda
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are functions where convention dictates that (ideally) a parameter
+should be called 'lambda'.  In Python, 'lambda' is a reserved word that
+cannot be used as a variable name.  If a new function or class has a
+parameter where a name that is similar to 'lambda' is really the best
+choice, the preferred name to use is [TBD: ``lam``, ``lmb``, ``lamb``,
+``lmbda``, ``lambda_``?].
+
+There has been a lack of consistency in the past:
+
+============================== =============
+Function or class              Variable name
+============================== =============
+`scipy.stats.boxcox`           ``lmbda``
+`scipy.stats.boxcox_llf`       ``lmb``
+`scipy.stats.yeojohnson`       ``lmbda``
+`scipy.stats.yeojohnson_llf`   ``lmb``
+`scipy.stats.chi2_contingency` ``lambda_``
+`scipy.stats.boltzmann`        ``lambda_``
+`scipy.stats.planck`           ``lambda_``
+`scipy.stats.tukeylambda`      ``lam``
+`scipy.signal.cspline1d`       ``lamb``
+`scipy.signal.qspline1d`       ``lamb``
+`scipy.signal.cspline2d`       ``lambda`` [*]
+`scipy.signal.qspline2d`       ``lambda`` [*]
+`scipy.signal.spline_filter`   ``lmbda``
+`scipy.special.tklmbda`        ``lmbda``
+`scipy.special.boxcox`         ``lmbda``
+`scipy.special.boxcox1p`       ``lmbda``
+`scipy.special.inv_boxcox`     ``lmbda``
+`scipy.special.inv_boxcox1p`   ``lmbda``
+============================== =============
+
+.. [*] This function accepts positional arguments only, but the docstring
+       documents the argument as `lambda`.
+
+
+Documentation Guidelines
+------------------------
+
+Use of LaTeX
+~~~~~~~~~~~~
+[TBD.  In the past, we had an unwritten rule that LaTex markup should be
+restricted to the Notes section.  We don't follow that rule these days.
+Should we?]
+
+
+Use "must", not "should"
+~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think we need even more detail here. The [numpydoc style guide](https://numpydoc.readthedocs.io/en/latest/format.html) offers only:
![image](https://user-images.githubusercontent.com/6570539/156815005-ace2d64e-5c1f-4fb6-b8f3-79074f2ce175.png)
![image](https://user-images.githubusercontent.com/6570539/156815115-436bc889-065a-4323-b622-bc17fa9118f5.png)

We should at least mention that function names should also have single backticks because they get linked. (And that function names from other sub-packages should be fully-qualified, e.g. `scipy.optimize.minimize`.)

Examples of the ambiguity that remains:

- Sometimes return values of functions are named in the documentation. Are these "variables" or "inline display"?
- What about strings or other values that are passed as the value of an argument (e.g. `'exact'`, `None`, `1.5`)?
- Parameters that are accepted in `kwargs` or similar? Named as keys of an `options` structure?

Then there's the question about _why_ it's appropriate for parameter names, which _are_ ultimately code, to appear sometimes in bold:
![image](https://user-images.githubusercontent.com/6570539/156816114-602349cf-5aa4-4e95-9c29-e45a4539f139.png)
sometimes in italics:
![image](https://user-images.githubusercontent.com/6570539/156816178-b498e325-2c88-474c-9be7-d714f9fb981e.png)
and sometimes as code (monospaced font / teletype):
![image](https://user-images.githubusercontent.com/6570539/156816268-2b91a83d-9d93-4743-98ac-8c97cc27d331.png)
(All of these are from [scipy.optimize.minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize), and all use single/double-backticks correctly, AFAICT.)

I'd suggest we can sidestep these questions and clarify the docs if all of these cases just appeared in monospace font, since they really are all code.

Reply:
And @tupui noted [above](https://github.com/scipy/scipy/pull/13955#issuecomment-836645230), we should specify how to refer to code in Errors and Warnings, too. (Of course, I'd suggest single backticks here, too.)